extensions/mssql/src/objectExplorerNodeProvider/connection.ts

```diff
@@ -6,226 +6,109 @@
 'use strict';
 
 import * as sqlops from 'sqlops';
-import * as UUID from 'vscode-languageclient/lib/utils/uuid';
 import * as nls from 'vscode-nls';
 const localize = nls.loadMessageBundle();
 
 import * as constants from '../constants';
-import * as LocalizedConstants from '../localizedConstants';
-import * as utils from '../utils';
 import { IFileSource, IHdfsOptions, IRequestParams, FileSourceFactory } from './fileSources';
-import { IEndpoint } from './objectExplorerNodeProvider';
 
-function appendIfExists(uri: string, propName: string, propValue: string): string {
-	if (propValue) {
-		uri = `${uri};${propName}=${propValue}`;
-	}
-	return uri;
-}
-
-interface IValidationResult {
-	isValid: boolean;
-	errors: string;
-}
-
-export class Connection {
+export class SqlClusterConnection {
+	private _connection: sqlops.connection.Connection;
+	private _profile: sqlops.IConnectionProfile;
 	private _host: string;
-	private _knoxPort: string;
-
-	constructor(private connectionInfo: sqlops.ConnectionInfo, private connectionUri?: string, private _connectionId?: string) {
-		if (!this.connectionInfo) {
-			throw new Error(localize('connectionInfoMissing', 'connectionInfo is required'));
-		}
-
-		if (!this._connectionId) {
-			this._connectionId = UUID.generateUuid();
+	private _port: string;
+	private _user: string;
+	private _password: string;
+
+	constructor(connectionInfo: sqlops.connection.Connection | sqlops.IConnectionProfile) {
+		this.validate(connectionInfo);
+
+		if ('id' in connectionInfo) {
+			this._profile = connectionInfo;
+			this._connection = this.toConnection(this._profile);
+		} else {
+			this._connection = connectionInfo;
+			this._profile = this.toConnectionProfile(this._connection);
```

Comment:
As previously mentioned this isn't actually used anywhere. You can remove all the toConnectionProfile functionality, this private variable, and the public accessor (unless I'm really missing something.

Reply:
@YurongHe can you check if this is actually used in the branch? Other than in the this.toConnection (which could be changed) I don't see a use. Why do we have it?